### DISCUSSION TITLE
Transcoding - possibility of disable creating resolution that make upscaling

### DISCUSSION THREAD

**Discussion by fcqpl** [OvenMediaLabs/OvenMediaEngine#1122] [Ideas]
Hello.
Could you please add something like "don't upscale video" in transcoding?
For example, if the user is broadcasting 1080p video, OME transcodes the image to 1080p, 720p and 480p;
user is broadcasting 720p - transcoding to 720p, 480p, and 360p;
user is broadcasting 480p - transcoding to 480p, 360p.
user is broadcasting 360p and lower - transcoding to 360p (no lower resolution declared in transcoder)

Currently when user is broadcasting 480p it get's upscaled to higher resolutions eg. 1080p.

Of course, the functionality will be useful when implemented ABR.

Something like: `<DoNotUpscale>`
```
<Video>
	<Codec>h264</Codec>
	<Width>1920</Width>
	<Height>1080</Height>
	<Bitrate>4000000</Bitrate>
	<Framerate>30</Framerate>
	<DoNotUpscale>true</DoNotUpscale>
</Video>
<Video>
	<Codec>h264</Codec>
	<Width>1280</Width>
	<Height>720</Height>
	<Bitrate>3000000</Bitrate>
	<Framerate>30</Framerate>
	<DoNotUpscale>true</DoNotUpscale>
</Video>
<Video>
	<Codec>h264</Codec>
	<Width>852</Width>
	<Height>480</Height>
	<Bitrate>2000000</Bitrate>
	<Framerate>30</Framerate>
	<DoNotUpscale>true</DoNotUpscale>
</Video>
<Video>
	<Codec>h264</Codec>
	<Width>640</Width>
	<Height>360</Height>
	<Bitrate>1000000</Bitrate>
	<Framerate>30</Framerate>
	<DoNotUpscale>false</DoNotUpscale>
</Video>
```

**Comment by Keukhan** [comment]
@fcqpl 

Thank you for the great idea.

The Prevent Upscaling option will be added to the future update plan.

Thanks

[a comment by an automated account is not shown]

**Comment by getroot** [comment]
This feature has been released. Thank you!

https://github.com/AirenSoft/OvenMediaEngine/issues/1053
